### PR TITLE
Nextflow: Decouple simple and batch squared job submission

### PR DESCRIPTION
**Pull request by kniec**
Prior to this PR the NF workshop had the following steps:

## CE & Queue in Batch 

1. create OD and Spot Compute Environment
1. create job and workflow queue

## Nextflow on Batch 

1. create Docker RNASeq image for the job
1. configure the nextflow with awsbatch as an executor (nextflow-runtime runs local on Cloud9)
1. run the simple version
1. create Docker Image for nextflow-runtime
1. create job-definition
1. submit nextflow-runtime to AWS Batch

Even when doing it myself I feel that this creates confusion because the workshop is not creating what is necessary for the next step first but setting up all the Batch CE and queues.
Thus, this PR saperates the flow into a nextflow201 and 202 step.

## Local run w/ AWS Batch

1. Spot Compute ENvironment
1. Job Queue 
1. Create RNASeq Docker image
1. Configure nextflow 

**The sequence in hugo is somehow not reflecting what it should be in my `compose up` test...** Please have a look out for the order there!

## AWS Batch Squared

1. Nextflow Docker Image
1. OD Compute Environment
1. Create Workflow Queue
1. Create Job Definition
1. Squared Run
1. Squared Debug

### CODE REVIEW

File: content/nextflow-on-aws-batch/35_nextflow201/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: "Local Run w/ AWS Batch"
+chapter: true
+weight: 35
+---
+
+## Nextflow on AWS Batch
+
+Now that we setup AWS Batch we can use Nextflow to submit jobs we are getting closer to our architecture.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/35_nextflow201/_index.md:9]: Now that we setup AWS Batch, **and that** we can use Nextflow to submit jobs, we are getting closer to our **target** architecture.

*Note the comas, cannot have them in bold for whatever reasons*

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md [hunk 2]
```diff
@@ -1,13 +1,12 @@
 ---
-title: "On-Demand"
+title: "OD Compute Environment"
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/31_od.md:2]: _I propose to keep the acronyms to minimum, especially in titles, even if covered before (not everyone reads everything)._
**On Demand** Compute Environment

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+chapter: false
+weight: 32
+---
+
+## Create Job Queue
+
+Again, let us created a queue via the console.
+![5_queue_workflow_0](/images/nextflow-on-aws-batch/batch/5_queue_workflow_0.png)
+
+To create the job queue we choose a name (**workflow-queue**) a priority of 1 and the map them to the correct compute environment (**od-ce**).
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md:12]: Change "priority of 1" to bold.

The goal is to let the attendees know in one glance where a keyboard input will be necessary.

File: content/nextflow-on-aws-batch/40_nextflow202/35_batch-squared-config.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Batch Sqared Config"
+title: "Create Job-Definition"
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/35_batch-squared-config.md:2]: No need for a dash '-' between Job and Definition

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+title: "Create Job Queues"
+chapter: false
+weight: 11
+---
+
+## Create Job Queue
+
+Let us created a queue via the console.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:9]: create not created

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+![5_queue_workflow_0](/images/nextflow-on-aws-batch/batch/5_queue_workflow_0.png)
+
+
+To create the job queue we choose a name (**job-queue**) a priority of 1 and the map them to the correct compute environment (**spot-ce**).
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:13]: Change "priority of 1" to bold.

The goal is to let the attendees know in one glance where a keyboard input will be necessary.

File: content/nextflow-on-aws-batch/35_nextflow201/_index.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+The local run will use the job-queue to submit jobs while the nextflow process is running locally.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/35_nextflow201/_index.md:11]: No need for dash '-' between job and queue.
And capitalize Nextflow

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md [hunk 8]
```diff
@@ -1,13 +1,12 @@
...
 chapter: false
 weight: 31
 ---
 
 ## Create EC2 On-Demand Compute Environment
 
-The nextflow tasks, which are supervising the orchestration of the actual workflow are not able to tolerate interruptions as they need to run for the duration of the complete workflow.
-Thus, we create a small On-Demand compute environment which will run the supervision task.
+Similar to what we did for the **spot-ce** used by the **job-queue** we are going to create a Compute Environment which uses a couple of large instances to run the nextflow process within a AWS Batch job.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/31_od.md:9]: Capitalize Nextflow

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+![5_queue_workflow_1](/images/nextflow-on-aws-batch/batch/5_queue_workflow_1.png)
+
+### Queue Dashboard
+
+The job queue dashboards shows the queue as available.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md:17]: dashboards show
or
dashboard shows

File: content/nextflow-on-aws-batch/35_nextflow201/_index.md (hunk 1 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/_index.md]: done

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md (hunk 2 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/31_od.md:2]: @alexis-turpin-aws problem is that it creates a line-break within the menue... :(
Do you still reckon we use the long version?

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md (hunk 3 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md:12]: good catch, done

File: content/nextflow-on-aws-batch/40_nextflow202/35_batch-squared-config.md (hunk 4 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/35_batch-squared-config.md]: done

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md (hunk 6 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:13]: done

File: content/nextflow-on-aws-batch/35_nextflow201/_index.md (hunk 7 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/_index.md:11]: done

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md (hunk 8 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/31_od.md:9]: done globally within the workshop, thanks!

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md (hunk 9 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md:17]: done

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md (hunk 5 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:9]: done

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md [hunk 10]
```diff
@@ -1,20 +1,19 @@
...
 weight: 31
 ---
 
 ## Create EC2 On-Demand Compute Environment
 
-The nextflow tasks, which are supervising the orchestration of the actual workflow are not able to tolerate interruptions as they need to run for the duration of the complete workflow.
-Thus, we create a small On-Demand compute environment which will run the supervision task.
+Similar to what we did for the **spot-ce** used by the **job-queue** we are going to create a Compute Environment which uses a couple of large instances to run the Nextflow process within a AWS Batch job.
 
 ### Setup Compute ENvironment
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/31_od.md:11]: Environment.

_I know it was not changed in this PR, but I like to live in danger_

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+---
+title: "Create Workflow Queues"
+chapter: false
+weight: 32
+---
+
+## Create Job Queue
+
+Again, let us created a queue via the console.
```
Comment (alexis-turpin-aws) [inline content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md]: create

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md (hunk 5 above)
Comment (alexis-turpin-aws) [reply content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:9]: Evidence shows that you lied or didn't commit :)

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md (hunk 2 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/31_od.md:2]: ok, deal

File: content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md (hunk 5 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/11_job_queue.md:9]: now for real

File: content/nextflow-on-aws-batch/35_nextflow201/15_create_rna_image.md [hunk 12]
```diff
@@ -61,7 +61,7 @@ EOF
 {{% notice info %}}
 If you are not familiar with [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds), please take a moment to let it sink in. :)
 What we are doing with the above Dockerfile is creating a `build` stage that installs all the dependencies to run `pip install` and subsequently installing `awscli`. Notice, that we use a different target.
-Without that pip would install everything in the already masive `/opt/conda/` directory. In the final stage we are setup up the environment to pick up `aws` and its libraries.
+Without that pip would install everything in the already masive `/opt/conda/` directory. In the final stage we copying over the pip-path of the build stage and setting up the environment to pick up `aws` and its libraries.
```
[the remark this answers is not in the document]
Comment (kniec) [reply content/nextflow-on-aws-batch/35_nextflow201/15_create_rna_image.md]: done

File: content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md (hunk 11 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/32_job_queue.md]: done

File: content/nextflow-on-aws-batch/40_nextflow202/31_od.md (hunk 10 above)
Comment (kniec) [reply content/nextflow-on-aws-batch/40_nextflow202/31_od.md:11]: done

**Pull request merged**